packages/batch/src/types.ts

```diff
@@ -4,18 +4,27 @@ import type {
   KinesisStreamRecord,
   SQSRecord,
 } from 'aws-lambda';
+import { SqsFifoPartialProcessor } from './SqsFifoPartialProcessor.js';
+import { BasePartialBatchProcessor } from './BasePartialBatchProcessor.js';
 
 /**
  * Options for batch processing
  *
+ * @template T The type of the batch processor, defaults to BasePartialBatchProcessor
  * @property context The context object provided by the AWS Lambda runtime
+ * @property skipGroupOnError The option to group on error during processing
  */
-type BatchProcessingOptions = {
+type BatchProcessingOptions<T = BasePartialBatchProcessor> = {
   /**
    * The context object provided by the AWS Lambda runtime. When provided,
    * it's made available to the handler function you specify
    */
   context: Context;
+  /**
```

Comment:
I think we can make this optional, so one doesn't have to pass it whenever they want to specify the other (also this way you can remove it from the unit tests)